src/shared/ensure.js

```diff
@@ -0,0 +1,19 @@
+import isArray from './isArray'
+import { isObject } from './typeof'
+
+export function ensureIsArray(arg, key) {
+  if (key && isObject(arg)) {
+    if (!isArray(arg[key])) {
+      arg[key] = []
+    }
+    return arg
+  } else {
```

Comment:
Invert if/else and using early return might improve code clarity here ☺️

Reply:
Well, its actually a bit of a strange function. I added it to ensure that the property of an object is an array as thats what I needed for allthose lifecycle hooks. Then I thought lets make it a bit more generic, even tough I dont need it.

This is the best I could make of it. We could change the `if(!isArray)` statement recursive (`arg[key] = ensureIsArray(arg[key])`) which might makes it more readable but I dont like that its costs an extra function call and/or that we always set the value for arg[key]

Reply:
From
```js
export function ensureIsArray(arg, key) {
  if (key && isObject(arg)) {
    if (!isArray(arg[key])) {
      arg[key] = []
    }
    return arg
  } else {
    return isArray(arg) ? arg : []
  }
}
```

to

```js
export function ensureIsArray(arg, key) {
  if (!key || !isObject(arg)) {
    return isArray(arg) ? arg : []
  }

  if (!isArray(arg[key])) {
    arg[key] = []
  }
  return arg
}
```

Reply:
Do you really think thats better readable then 4968ee10b92864b0dd8e4eb7a3d52612553d767a? In my mind the `||` with the inline if makes me have to read that part twice to understand whats happening, while 4968ee10b92864b0dd8e4eb7a3d52612553d767a reads much more fluently.

Reply:
@pimlie `!(key && isObject(arg))` would be fine for me as well. :relaxed:

Reply:
Yeah but thats goes against the principle that `if(true)` provides better readability then `if(!false)` ;)

I have changed it to your proposal, even though I dont think its more readable I also dont care enough to have lengthy discussions about it ;)